pkg/imgpkg/bundle/images_lock.go

```diff
@@ -66,20 +66,49 @@ func (i ImageRefs) DeepCopy() ImageRefs {
 	return result
 }
 
-func NewImagesLock(imagesLock lockconfig.ImagesLock, imgRetriever ctlimg.ImagesMetadata, relativeToRepo string) *ImagesLock {
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . ImagesLockLocationConfig
+type ImagesLockLocationConfig interface {
+	Fetch() (ImageLocationsConfig, error)
+}
+
+func NewImagesLock(imagesLock lockconfig.ImagesLock, imgRetriever ctlimg.ImagesMetadata, relativeToRepo string, imagesLockLocationConfig ImagesLockLocationConfig) *ImagesLock {
 	imageRefs := ImageRefs{}
 	for _, image := range imagesLock.Images {
 		imageRefs.AddImagesRef(ImageRef{ImageRef: image, IsBundle: nil})
 	}
 
-	imgsLock := &ImagesLock{imageRefs: imageRefs, imgRetriever: imgRetriever}
+	imgsLock := &ImagesLock{imageRefs: imageRefs, imgRetriever: imgRetriever, imagesLockLocationFetcher: imagesLockLocationConfig}
 	imgsLock.generateImagesLocations(relativeToRepo)
+
 	return imgsLock
 }
 
+func (o *ImagesLock) SyncImageRefs() error {
```

Comment:
it seems that this method is mostly used to mutate ImagesLock with isBundle information. from the design perspective it seems quite surprising that i need to call this method myself, instead of ImagesLock managing such information internally (aka internal detail is leaking out to consumers). additionally i would have expected that we would be recording that image is at a particular location (that's why we introduced locations in the first place), but i dont see happening.

Reply:
as discussed over zoom, i moved that function into `ImageRefs`